// Code generated by mockery v0.0.0-dev. DO NOT EDIT.

package mocks

import (
	context "context"

	accounts "github.com/erickoliv/finances-api/accounts"

	mock "github.com/stretchr/testify/mock"

	rest "github.com/erickoliv/finances-api/pkg/http/rest"

	uuid "github.com/google/uuid"
)

// Repository is an autogenerated mock type for the Repository type
type Repository struct {
	mock.Mock
}

// Delete provides a mock function with given fields: ctx, pk, owner
func (_m *Repository) Delete(ctx context.Context, pk uuid.UUID, owner uuid.UUID) error {
	ret := _m.Called(ctx, pk, owner)

	var r0 error
	if rf, ok := ret.Get(0).(func(context.Context, uuid.UUID, uuid.UUID) error); ok {
		r0 = rf(ctx, pk, owner)
	} else {
		r0 = ret.Error(0)
	}

	return r0
}

// Get provides a mock function with given fields: ctx, pk, owner
func (_m *Repository) Get(ctx context.Context, pk uuid.UUID, owner uuid.UUID) (*accounts.Account, error) {
	ret := _m.Called(ctx, pk, owner)

	var r0 *accounts.Account
	if rf, ok := ret.Get(0).(func(context.Context, uuid.UUID, uuid.UUID) *accounts.Account); ok {
		r0 = rf(ctx, pk, owner)
	} else {
		if ret.Get(0) != nil {
			r0 = ret.Get(0).(*accounts.Account)
		}
	}

	var r1 error
	if rf, ok := ret.Get(1).(func(context.Context, uuid.UUID, uuid.UUID) error); ok {
		r1 = rf(ctx, pk, owner)
	} else {
		r1 = ret.Error(1)
	}

	return r0, r1
}

// Query provides a mock function with given fields: ctx, filters
func (_m *Repository) Query(ctx context.Context, filters *rest.Query) ([]*accounts.Account, error) {
	ret := _m.Called(ctx, filters)

	var r0 []*accounts.Account
	if rf, ok := ret.Get(0).(func(context.Context, *rest.Query) []*accounts.Account); ok {
		r0 = rf(ctx, filters)
	} else {
		if ret.Get(0) != nil {
			r0 = ret.Get(0).([]*accounts.Account)
		}
	}

	var r1 error
	if rf, ok := ret.Get(1).(func(context.Context, *rest.Query) error); ok {
		r1 = rf(ctx, filters)
	} else {
		r1 = ret.Error(1)
	}

	return r0, r1
}

// Save provides a mock function with given fields: ctx, account
func (_m *Repository) Save(ctx context.Context, account *accounts.Account) error {
	ret := _m.Called(ctx, account)

	var r0 error
	if rf, ok := ret.Get(0).(func(context.Context, *accounts.Account) error); ok {
		r0 = rf(ctx, account)
	} else {
		r0 = ret.Error(0)
	}

	return r0
}
